apps/meteor/server/models/raw/Rooms.ts

```diff
@@ -868,6 +868,27 @@ export class RoomsRaw extends BaseRaw<IRoom> implements IRoomsModel {
 		return this.col.countDocuments({ federated: true });
 	}
 
+	async countMessagesInActiveFederatedRooms(): Promise<number> {
+		const pipeline = [
+			{
+				$match: {
+					federated: true,
```

Comment:
although we already have a `countFederatedRooms` which filters by this field, we're missing an index here.. I'd suggest we remove this one from this PR and add it back in another one.